chainer/functions/array/hstack.py

```diff
@@ -53,11 +53,31 @@ def hstack(xs):
     """Concatenate variables horizontally (column wise).
 
     Args:
-        xs (list of chainer.Variable): Variables to be concatenated.
+        xs (list of :class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variables to be concatenated. The variables must have the
+            same shape, except in the horizontal (column wise) dimension.
```

Comment:
Thank you for investigation. In your case, `x2.ndim` equals to `2`, not `0` nor `1` (`ndim` is a dimension of the ndarray, which is equal to `len(x.shape)`). As `hstack` concatenate input arrays along second axes, it is not trivial for users what happens if input arrays do not have the axes.